docs/ParameterValidation.md

```diff
@@ -0,0 +1,177 @@
+    - Content-Type: text/plain
+    - Response Body: Invalid actorId parameter
+
+- Name: genre
+- Type: string
+- Parameter Validation:
+  - Valid input: case insensitive genre,length [3,20]
+    - Example: 'War','war','Documentary','documentary'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies filtered by genre as [{},{},..],empty array when no results,where movie.genres contains genre
+  - Invalid input: genre length < 3 or >20 characters
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid genre parameter
+
+- Name: pageSize
+- Type: integer
+- Description: pageSize limits the number of results returned from the server to a specified number of results
+- Parameter Validation:
+  - Default value: 100
+  - Valid input range: [1,1000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies limited to pageSize as [{},{},..], empty array when no results
+  - Invalid input: input that does not parse or is out of range
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageSize parameter
+
+- Name: pageNumber
+- Type: integer
+- Description: pageNumber sets the first position to return from the results of the query
+- Parameter Validation:
+  - Default value: 1
+  - Valid input range: [1,10000]
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: JSON Array of movies based on pageNumber as [{},{},..],empty array when no results
+  - Invalid input: input that does not parse or is out of range
+    - Example: 100.23
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid pageNumber parameter
+
+### Movies API Parameters
+
+- API /api/movies/{movieId}
+- Name: movieId
+- Type: string
+- Parameter Validation:
+  - Valid input: well-formed movieId rule - starts with lower case 'tt' followed by 5-9 digits,parsed as an int. This is case sensitive input
+    - Example 'tt0114746'
+    - Status: 200
+    - Content-Type: application/json
+    - Response Body: Single movie document as JSON
+  - Valid input: movieId does not exist
+    - Example: 'tt123456'
+    - Status: 404
+    - Content-Type: text/plain
+    - Response Body: movie not found
+  - Invalid input: movieId not conforming to well-formed movieId rule
+    - Example:'tt1234','nm1234','TT123456','nm1265067'
+    - Status: 400
+    - Content-Type: text/plain
+    - Response Body: Invalid movieId parameter
+
+## Actors
+
+### API /api/actors
+
+### Actors Query Parameters
+
+- Query parameters for actors are q,pageSize,pageNumber
+- [Refer to the rules for Query Parameters for Movies](###movies-query-parameters)
+
+### Actors API Parameters
+
+- API /api/actors/{actorId}
+- Name: actorId
+- Type: string
+- Parameter Validation:
+  - Valid input: well-formed actorId rule - starts with 'nm' followed by 5-9 digits,parsed as an int
+    - Example: 'nm1265067','NM123456'
```

Comment:
This is a different rule than the actorId parameter mentioned earlier.  I think this should require the 'nm' to be lowercase to be valid to match the other id parameters in this doc (actorId and movieId).

Reply:
yes, NM123456 and TT123456 are invalid - sorry I missed that

Reply:
Yes, updated these.